examples/plot_complex_data.py

```diff
@@ -0,0 +1,191 @@
+# %%
+import numpy as np
+import hazardous.data._competing_weibull as competing_w
+from time import perf_counter
+import matplotlib.pyplot as plt
+from sklearn.model_selection import train_test_split
+import pandas as pd
+import seaborn as sns
+
+from hazardous import GradientBoostingIncidence
+from lifelines import AalenJohansenFitter
+
+seed = 0
+rng = np.random.RandomState(seed)
+DEFAULT_SHAPE_RANGES = (
+    (0.7, 0.9),
+    (1.0, 1.0),
+    (2.0, 3.0),
+)
+
+DEFAULT_SCALE_RANGES = (
+    (1, 20),
+    (1, 10),
+    (1.5, 5),
+)
+n_events = 3
+
+
+# %%
+# In the following cell, we verify that the synthetic dataset is well defined.
+
+rng = np.random.RandomState(seed)
+df_features = pd.DataFrame(rng.randn(100_000, 10))
+df_features.columns = [f"feature_{i}" for i in range(10)]
+
+df_shape_scale_star = competing_w.compute_shape_and_scale(
+    df_features,
+    6,
+    0.3,
+    n_events,
+    2,
+    DEFAULT_SHAPE_RANGES,
+    DEFAULT_SCALE_RANGES,
+    0,
```

Comment:
Please pass the name of all the keyword arguments to ease understanding of the example by people not familiar with the API.